blosc/shuffle-altivec.c

```diff
@@ -15,61 +16,145 @@
-          buf[12], buf[13], buf[14], buf[15]);
+    //Load the surrounding area
+    __vector uint8_t low = vec_ld( position, dst);
+    __vector uint8_t high = vec_ld( position+16, dst);
+    //Prepare the constants that we need
+    uint8_t offset = ((size_t)dst) % 16;
+    __vector uint8_t permute_lo, permute_hi;
+    for (uint8_t i=0; i<16; i++)
+    {
+        if (i<offset){
+            permute_lo[i] = i;
+            permute_hi[i] = 32 + i - offset;
+        }
+        else{
+            permute_lo[i] = 16 + i - offset;
+            permute_hi[i] = i;
+        }
+    }
+    //Insert our data into the low and high vectors
+    low = vec_perm( low, v, permute_lo );
+    high = vec_perm( high, v, permute_hi );
+    //Store the two aligned result vectors
+    vec_st( low, position, dst);
+    vec_st( high, position+16, dst);
 }
-#endif
-
+/* Missaligned storage helper function which su
+ * 
+ * param:
+ *     data: the (misaligned) vector to store
+ *     position: index in dst where to store
+ *     dst: the pointer of where to store
+ *     previous: the remaining vector of the last write
+ *     shuffle_lo: the vector for shuffling the lower part
+ *     shuffle_hi: the vector for shuffling the higher part
+ * 
+ * return: The remainder to be still written in next vector filled with zeros
+ */
+static inline __vector uint8_t helper_misaligned_store(__vector uint8_t data, 
+                                                       int32_t position,  
+                                                       uint8_t * dst,
+                                                       __vector uint8_t previous,
+                                                       __vector uint8_t shuffle_lo,
+                                                       __vector uint8_t shuffle_hi)
+{  
+    static const __vector uint8_t zero = (const __vector uint8_t) {0, 0, 0, 0, 0, 0, 0, 0,
+                                                                   0, 0, 0, 0, 0, 0, 0, 0};
+    __vector uint8_t low, high;
+    //Insert our data into the low and high vectors
+    low = vec_perm( previous, data, shuffle_lo );
+    high = vec_perm( zero, data, shuffle_hi );
+    //Store the two aligned result vectors
+    vec_st( low, position, dst);
+    // Return the next "previous" vector
+    return high;
+}                                                 
+
+/* Calculate the permutation vector for storing data in the lower part
+ * 
+ * param: offset, the position
+ */
+static inline __vector uint8_t gen_permute_low(int32_t offset)
+{
+    __vector uint8_t permute;
+    for (uint8_t i=0; i<16; i++)
+    {
+       if (i<offset) permute[i] = i;
+       else permute[i] = 16 + i - offset;
+    }
+    return permute;
+} 
+
+/* Calculate the permutation vector for storing data in the upper part
+ * 
+ * param: offset, the position
+ */
+static inline __vector uint8_t gen_permute_high(int32_t offset)
+{
+    __vector uint8_t permute;
+    for (int32_t i=0; i<16; i++)
```

Comment:
Use spaces between variables and assignements or comparisons, just like Python does.